Add SOCIAL_AUTH_BACKEND_PREFIX for custom auth backends (#807)
nautobot/nautobot: nautobot/docs/configuration/authentication/sso.md

> To use external authentcation, you'll need to define `AUTHENTICATION_BACKENDS` in your `nautobot_config.py`.

Reviewer: Since you're in here anyway, mind fixing this? :-)

```suggestion
To use external authentication, you'll need to define `AUTHENTICATION_BACKENDS` in your `nautobot_config.py`.
```

!!! warning
	You should only enable one social authentication authentication backend. It is technically possible to use multiple backends but we cannot officially support more than one at this time.

### Custom Authentication Backends

> The default external authentication supported is [social-auth-app-django](https://python-social-auth.readthedocs.io/en/latest/configuration/django.html) as stated above. If you have developed your own external authentication backend, you can specify `SOCIAL_AUTH_BACKEND_PREFIX` to enable the SSO redirect when the login button is clicked. Currently, Nautobot will look for `social_core.backends` within `AUTHENTICATION_BACKENDS` to endable the redirect.Let's take a look at an example for a custom authentication backend available at `custom_auth.backends.custom.Oauth2`.

Reviewer: ```suggestion
The default external authentication supported is [social-auth-app-django](https://python-social-auth.readthedocs.io/en/latest/configuration/django.html) as stated above. If you have developed your own external authentication backend, you will need to configure `SOCIAL_AUTH_BACKEND_PREFIX` to use your backend instead and correctly enable the SSO redirect when the login button is clicked. For example, if your custom authentication backend is available at `custom_auth.backends.custom.Oauth2`, you would set things as follows:
```

```python

> SOCIAL_AUTH_BACKEND_PREFIX = "custom_auth.backends"

Reviewer: ```suggestion
SOCIAL_AUTH_BACKEND_PREFIX = "custom_auth.backends"

AUTHENTICATION_BACKENDS = [
    "custom_auth.backends.custom.Oauth2",
    "nautobot.core.authentication.ObjectPermissionBackend",
]
```

```

> `SOCIAL_AUTH_BACKEND_PREFIX` was set to `custom_auth.backends` as we may have more than one custom backend. This will enable the SSO redirect for users when they click the login button.

Reviewer: I'm not quite clear what the first sentence here is saying?

Author: I will have this updated once my local tests pass and I push up. Hopefully it's more clear, but let me know.

| [Google](https://python-social-auth.readthedocs.io/en/latest/backends/google.html)                            | `social_core.backends.gae.GoogleAppEngineAuth`                  |
|                                                                                                               | `social_core.backends.google.GoogleOAuth2`                      |
|                                                                                                               | `social_core.backends.google.GoogleOAuth`                       |
|                                                                                                               | `social_core.backends.google_openidconnect.GoogleOpenIdConnect` |
| [Okta](https://python-social-auth.readthedocs.io/en/latest/backends/okta.html)                                | `social_core.backends.okta.OktaOAuth2`                          |
|                                                                                                               | `social_core.backends.okta_openidconnect.OktaOpenIdConnect`     |

> | [SAML](https://python-social-auth.readthedocs.io/en/latest/backends/saml.html)                                | `social_core.backends.saml.SAMLAuth`                            |

Reviewer: I personally am not in favor of this change - it doesn't improve readability IMO.

Author: I will get this changed. It was my auto formatter x.x
